### PR TITLE
aws_fsx_lustre_file_system support Persistent 2, add new resource aws_fsx_data_repository_association

### LINKED ISSUE #22279: Support `PERSISTENT_2` in `aws_fsx_lustre_file_system`
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The `aws_fsx_lustre_file_system` resource currently supports the following values for `deployment_type`:

- `SCRATCH_1`
- `SCRATCH_2`
- `PERSISTENT_1`

As of approximately November 2021, AWS introduced the `PERSISTENT_2` deployment type. This introduces a number of new features in FSx for Lustre, including the ability to create many data repository associations for a given file system. Unlike earlier `PERSISTENT_1` data repositories, these associations can be configured to export data back to the source S3 bucket.[^1]

The current resource has `import_path` and `export_path` attributes. These would not be applicable to a `PERSISTENT_2` deployment. For `PERSISTENT_2` file system, its repository associations would need to be configured after the file system is created.

[^1]: https://awscli.amazonaws.com/v2/documentation/api/latest/reference/fsx/create-data-repository-association.html

### New or Affected Resource(s)

* aws_fsx_lustre_file_system
* aws_fsx_lustre_data_repository_association (new)

### Potential Terraform Configuration

```hcl
resource "aws_fsx_lustre_file_system" "example" {
  storage_capacity = 1200
  subnet_ids       = [aws_subnet.example.id]
  deployment_type  = "PERSISTENT_2"
}

resource "aws_fsx_lustre_data_repository_association" "example" {
  file_system_id       = aws_fsx_lustre_file_system.example.id
  file_system_path     = "/example"
  data_repository_path = "s3://${aws_s3_bucket.example.bucket}"
  
  s3 {
    auto_import_policy = "NEW_CHANGED_DELETED"
    auto_export_policy = "NEW_CHANGED_DELETED"
  }  
}
```

### References

* [API Reference: Create Data Repository Association](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/fsx/create-data-repository-association.html)
* [Documentation Revision 2021-11-30](https://docs.aws.amazon.com/fsx/latest/LustreGuide/doc-history.html)
* [Documentation: Persistent file systems](https://docs.aws.amazon.com/fsx/latest/LustreGuide/using-fsx-lustre.html#persistent-file-system)

### PR DESCRIPTION
**Pull request by vicyap**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

### About

* Updates `aws_fsx_lustre_file_system` to support `PERSISTENT_2` as a deployment type.
* Adds a new resource, `aws_fsx_data_repository_association` to link S3 buckets to FSx for Lustre Filesystems using Persistent 2.

Docs: https://docs.aws.amazon.com/fsx/latest/LustreGuide/overview-dra-data-repo.html

Closes #22279

I probably missed a few things. Please let me know anything I should address!

Output from acceptance testing:

```
$ make testacc ACCTEST_PARALLELISM=10 PKG=fsx TESTS='TestAccFSxDataRepositoryAssociation_*|TestAccFSxLustreFileSystem_deploymentTypePersistent2|TestAccFSxLustreFileSystem_fileSystemTypeVersion'

==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/fsx/... -v -count 1 -parallel 10 -run='TestAccFSxDataRepositoryAssociation_*|TestAccFSxLustreFileSystem_deploymentTypePersistent2|TestAccFSxLustreFileSystem_fileSystemTypeVersion' -timeout 180m
=== RUN   TestAccFSxDataRepositoryAssociation_basic
=== PAUSE TestAccFSxDataRepositoryAssociation_basic
=== RUN   TestAccFSxDataRepositoryAssociation_disappears
=== PAUSE TestAccFSxDataRepositoryAssociation_disappears
=== RUN   TestAccFSxDataRepositoryAssociation_disappears_ParentFileSystem
=== PAUSE TestAccFSxDataRepositoryAssociation_disappears_ParentFileSystem
=== RUN   TestAccFSxDataRepositoryAssociation_fileSystemPathUpdated
=== PAUSE TestAccFSxDataRepositoryAssociation_fileSystemPathUpdated
=== RUN   TestAccFSxDataRepositoryAssociation_dataRepositoryPathUpdated
=== PAUSE TestAccFSxDataRepositoryAssociation_dataRepositoryPathUpdated
=== RUN   TestAccFSxDataRepositoryAssociation_importedFileChunkSize
=== PAUSE TestAccFSxDataRepositoryAssociation_importedFileChunkSize
=== RUN   TestAccFSxDataRepositoryAssociation_importedFileChunkSizeUpdated
=== PAUSE TestAccFSxDataRepositoryAssociation_importedFileChunkSizeUpdated
=== RUN   TestAccFSxDataRepositoryAssociation_deleteDataInFilesystem
=== PAUSE TestAccFSxDataRepositoryAssociation_deleteDataInFilesystem
=== RUN   TestAccFSxDataRepositoryAssociation_s3AutoExportPolicy
=== PAUSE TestAccFSxDataRepositoryAssociation_s3AutoExportPolicy
=== RUN   TestAccFSxDataRepositoryAssociation_s3AutoExportPolicyUpdate
=== PAUSE TestAccFSxDataRepositoryAssociation_s3AutoExportPolicyUpdate
=== RUN   TestAccFSxDataRepositoryAssociation_s3AutoImportPolicy
=== PAUSE TestAccFSxDataRepositoryAssociation_s3AutoImportPolicy
=== RUN   TestAccFSxDataRepositoryAssociation_s3AutoImportPolicyUpdate
=== PAUSE TestAccFSxDataRepositoryAssociation_s3AutoImportPolicyUpdate
=== RUN   TestAccFSxDataRepositoryAssociation_s3FullPolicy
=== PAUSE TestAccFSxDataRepositoryAssociation_s3FullPolicy
=== RUN   TestAccFSxLustreFileSystem_fileSystemTypeVersion
=== PAUSE TestAccFSxLustreFileSystem_fileSystemTypeVersion
=== RUN   TestAccFSxLustreFileSystem_deploymentTypePersistent2
=== PAUSE TestAccFSxLustreFileSystem_deploymentTypePersistent2
=== CONT  TestAccFSxDataRepositoryAssociation_basic
=== CONT  TestAccFSxDataRepositoryAssociation_s3AutoExportPolicy
=== CONT  TestAccFSxDataRepositoryAssociation_s3AutoImportPolicyUpdate
=== CONT  TestAccFSxDataRepositoryAssociation_dataRepositoryPathUpdated
=== CONT  TestAccFSxDataRepositoryAssociation_s3AutoExportPolicyUpdate
=== CONT  TestAccFSxLustreFileSystem_deploymentTypePersistent2
=== CONT  TestAccFSxLustreFileSystem_fileSystemTypeVersion
=== CONT  TestAccFSxDataRepositoryAssociation_s3AutoImportPolicy
=== CONT  TestAccFSxDataRepositoryAssociation_s3FullPolicy
=== CONT  TestAccFSxDataRepositoryAssociation_deleteDataInFilesystem
--- PASS: TestAccFSxLustreFileSystem_deploymentTypePersistent2 (531.51s)
=== CONT  TestAccFSxDataRepositoryAssociation_disappears_ParentFileSystem
--- PASS: TestAccFSxDataRepositoryAssociation_deleteDataInFilesystem (673.84s)
=== CONT  TestAccFSxDataRepositoryAssociation_fileSystemPathUpdated
--- PASS: TestAccFSxDataRepositoryAssociation_dataRepositoryPathUpdated (814.48s)
=== CONT  TestAccFSxDataRepositoryAssociation_disappears
--- PASS: TestAccFSxDataRepositoryAssociation_basic (898.54s)
=== CONT  TestAccFSxDataRepositoryAssociation_importedFileChunkSize
--- PASS: TestAccFSxDataRepositoryAssociation_s3AutoExportPolicy (942.60s)
=== CONT  TestAccFSxDataRepositoryAssociation_importedFileChunkSizeUpdated
--- PASS: TestAccFSxDataRepositoryAssociation_s3AutoImportPolicy (1046.42s)
--- PASS: TestAccFSxLustreFileSystem_fileSystemTypeVersion (1127.75s)
--- PASS: TestAccFSxDataRepositoryAssociation_disappears_ParentFileSystem (606.02s)
--- PASS: TestAccFSxDataRepositoryAssociation_s3AutoExportPolicyUpdate (1160.87s)
--- PASS: TestAccFSxDataRepositoryAssociation_s3FullPolicy (1346.61s)
--- PASS: TestAccFSxDataRepositoryAssociation_s3AutoImportPolicyUpdate (1491.98s)
--- PASS: TestAccFSxDataRepositoryAssociation_fileSystemPathUpdated (838.60s)
--- PASS: TestAccFSxDataRepositoryAssociation_importedFileChunkSize (651.49s)
--- PASS: TestAccFSxDataRepositoryAssociation_disappears (741.27s)
--- PASS: TestAccFSxDataRepositoryAssociation_importedFileChunkSizeUpdated (772.72s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/fsx        1718.381s
```

### CODE REVIEW

File: internal/provider/provider.go [hunk 1]
```diff
@@ -1215,6 +1215,7 @@ func Provider() *schema.Provider {
 
 			"aws_fsx_backup":                        fsx.ResourceBackup(),
 			"aws_fsx_lustre_file_system":            fsx.ResourceLustreFileSystem(),
+			"aws_fsx_data_repository_association":   fsx.ResourceDataRepositoryAssociation(),
```
Comment (vicyap) [inline internal/provider/provider.go:1218]: Should it be called ‘aws_fsx_data_repository_association’ or ‘aws_fsx_lustre_data_repository_association’?

File: website/docs/r/fsx_data_repository_association.html.markdown [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+subcategory: "File System (FSx)"
+layout: "aws"
+page_title: "AWS: aws_fsx_data_repository_association"
+description: |-
+  Manages a FSx for Lustre Data Repository Association.
+---
+
+# Resource: aws_fsx_data_repository_association
+
+Manages a Manages a FSx for Lustre Data Repository Association.. See [Linking your file system to an S3 bucket](https://docs.aws.amazon.com/fsx/latest/LustreGuide/create-dra-linked-data-repo.html) for more information.
```
Comment (jrykr) [inline website/docs/r/fsx_data_repository_association.html.markdown:11]: Typo to fix

**Review by jrykr — COMMENTED** [review]
Awesome! Thanks.

File: website/docs/r/fsx_data_repository_association.html.markdown (hunk 2 above)
Comment (vicyap) [reply website/docs/r/fsx_data_repository_association.html.markdown]: addressed

**Comment by vicyap** [comment]
@jrykr I updated the PR with a typo fix.

What other changes do I need to make?

**Comment by jrykr** [comment]
@vicyap, lgtm, but I'm not a maintainer. 

@DrFaust92, would you be able to help get this PR triaged?

**Comment by vicyap** [comment]
Hello @DrFaust92, are there any updates on this PR? I understand this project is full of PRs to be reviewed. Any estimate on when this will get reviewed and merged would be helpful.

**Review by DrFaust92 — APPROVED** [review]
cant verify tests at the moment but looks great!

File: .changelog/22291.txt [hunk 3]
```diff
@@ -0,0 +1,7 @@
+```release-note:enhancement
+resource/aws_fsx_lustre_file_system: Support PERSISTENT_2 for deployement_type
```
Comment (DrFaust92) [inline .changelog/22291.txt:2]: lets add another entry for adding `file_system_type_version`

**Comment by breathingdust** [comment]
Hey @DrFaust92! I can validate tests tonight.

**Comment by breathingdust** [comment]
Verified Tests in Commercial (us-west-2)

```console
make testacc ACCTEST_PARALLELISM=10 PKG=fsx TESTS='TestAccFSxDataRepositoryAssociation_*|TestAccFSxLustreFileSystem_deploymentTypePersistent2|TestAccFSxLustreFileSystem_fileSystemTypeVersion'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/fsx/... -v -count 1 -parallel 10 -run='TestAccFSxDataRepositoryAssociation_*|TestAccFSxLustreFileSystem_deploymentTypePersistent2|TestAccFSxLustreFileSystem_fileSystemTypeVersion' -timeout 180m
=== RUN   TestAccFSxDataRepositoryAssociation_basic
=== PAUSE TestAccFSxDataRepositoryAssociation_basic
=== RUN   TestAccFSxDataRepositoryAssociation_disappears
=== PAUSE TestAccFSxDataRepositoryAssociation_disappears
=== RUN   TestAccFSxDataRepositoryAssociation_disappears_ParentFileSystem
=== PAUSE TestAccFSxDataRepositoryAssociation_disappears_ParentFileSystem
=== RUN   TestAccFSxDataRepositoryAssociation_fileSystemPathUpdated
=== PAUSE TestAccFSxDataRepositoryAssociation_fileSystemPathUpdated
=== RUN   TestAccFSxDataRepositoryAssociation_dataRepositoryPathUpdated
=== PAUSE TestAccFSxDataRepositoryAssociation_dataRepositoryPathUpdated
=== RUN   TestAccFSxDataRepositoryAssociation_importedFileChunkSize
=== PAUSE TestAccFSxDataRepositoryAssociation_importedFileChunkSize
=== RUN   TestAccFSxDataRepositoryAssociation_importedFileChunkSizeUpdated
=== PAUSE TestAccFSxDataRepositoryAssociation_importedFileChunkSizeUpdated
=== RUN   TestAccFSxDataRepositoryAssociation_deleteDataInFilesystem
=== PAUSE TestAccFSxDataRepositoryAssociation_deleteDataInFilesystem
=== RUN   TestAccFSxDataRepositoryAssociation_s3AutoExportPolicy
=== PAUSE TestAccFSxDataRepositoryAssociation_s3AutoExportPolicy
=== RUN   TestAccFSxDataRepositoryAssociation_s3AutoExportPolicyUpdate
=== PAUSE TestAccFSxDataRepositoryAssociation_s3AutoExportPolicyUpdate
=== RUN   TestAccFSxDataRepositoryAssociation_s3AutoImportPolicy
=== PAUSE TestAccFSxDataRepositoryAssociation_s3AutoImportPolicy
=== RUN   TestAccFSxDataRepositoryAssociation_s3AutoImportPolicyUpdate
=== PAUSE TestAccFSxDataRepositoryAssociation_s3AutoImportPolicyUpdate
=== RUN   TestAccFSxDataRepositoryAssociation_s3FullPolicy
=== PAUSE TestAccFSxDataRepositoryAssociation_s3FullPolicy
=== RUN   TestAccFSxLustreFileSystem_fileSystemTypeVersion
=== PAUSE TestAccFSxLustreFileSystem_fileSystemTypeVersion
=== RUN   TestAccFSxLustreFileSystem_deploymentTypePersistent2
=== PAUSE TestAccFSxLustreFileSystem_deploymentTypePersistent2
=== CONT  TestAccFSxDataRepositoryAssociation_basic
=== CONT  TestAccFSxDataRepositoryAssociation_s3AutoExportPolicy
=== CONT  TestAccFSxDataRepositoryAssociation_dataRepositoryPathUpdated
=== CONT  TestAccFSxDataRepositoryAssociation_s3FullPolicy
=== CONT  TestAccFSxLustreFileSystem_deploymentTypePersistent2
=== CONT  TestAccFSxLustreFileSystem_fileSystemTypeVersion
=== CONT  TestAccFSxDataRepositoryAssociation_s3AutoImportPolicy
=== CONT  TestAccFSxDataRepositoryAssociation_disappears_ParentFileSystem
=== CONT  TestAccFSxDataRepositoryAssociation_s3AutoImportPolicyUpdate
=== CONT  TestAccFSxDataRepositoryAssociation_fileSystemPathUpdated
--- PASS: TestAccFSxLustreFileSystem_deploymentTypePersistent2 (585.71s)
=== CONT  TestAccFSxDataRepositoryAssociation_s3AutoExportPolicyUpdate
--- PASS: TestAccFSxDataRepositoryAssociation_basic (715.42s)
=== CONT  TestAccFSxDataRepositoryAssociation_importedFileChunkSizeUpdated
--- PASS: TestAccFSxDataRepositoryAssociation_fileSystemPathUpdated (730.29s)
=== CONT  TestAccFSxDataRepositoryAssociation_deleteDataInFilesystem
--- PASS: TestAccFSxDataRepositoryAssociation_dataRepositoryPathUpdated (792.95s)
=== CONT  TestAccFSxDataRepositoryAssociation_importedFileChunkSize
--- PASS: TestAccFSxDataRepositoryAssociation_disappears_ParentFileSystem (906.53s)
=== CONT  TestAccFSxDataRepositoryAssociation_disappears
--- PASS: TestAccFSxDataRepositoryAssociation_s3AutoExportPolicy (941.10s)
--- PASS: TestAccFSxDataRepositoryAssociation_s3FullPolicy (1207.10s)
--- PASS: TestAccFSxDataRepositoryAssociation_s3AutoImportPolicy (1217.60s)
--- PASS: TestAccFSxLustreFileSystem_fileSystemTypeVersion (1246.37s)
--- PASS: TestAccFSxDataRepositoryAssociation_deleteDataInFilesystem (671.86s)
--- PASS: TestAccFSxDataRepositoryAssociation_importedFileChunkSize (657.77s)
--- PASS: TestAccFSxDataRepositoryAssociation_importedFileChunkSizeUpdated (755.33s)
--- PASS: TestAccFSxDataRepositoryAssociation_s3AutoImportPolicyUpdate (1512.04s)
--- PASS: TestAccFSxDataRepositoryAssociation_s3AutoExportPolicyUpdate (961.22s)
--- PASS: TestAccFSxDataRepositoryAssociation_disappears (953.82s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/fsx	1863.510s
```

Hi @vicyap 👋. The tests are failing in GovCloud as Persistent2 is not available in that partition. Typically what we do is modify the tests so they they skip rather than fail when run there. We will follow up with that change ourselves as its unlikely you have a GovCloud account to test in!

Looks like this is almost GTG, just need some formatting fixes to get the Acceptance Test Linting check to go green and the CHANGELOG entry @DrFaust92 mentioned and we can merge!

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

##### Commercial

```console
% make testacc TESTARGS='-run=TestAccFSxDataRepositoryAssociation_\|TestAccFSxLustreFileSystem_' PKG=fsx ACCTEST_PARALLELISM=10
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/fsx/... -v -count 1 -parallel 10 -run=TestAccFSxDataRepositoryAssociation_\|TestAccFSxLustreFileSystem_ -timeout 180m
=== RUN   TestAccFSxDataRepositoryAssociation_basic
=== PAUSE TestAccFSxDataRepositoryAssociation_basic
=== RUN   TestAccFSxDataRepositoryAssociation_disappears
=== PAUSE TestAccFSxDataRepositoryAssociation_disappears
=== RUN   TestAccFSxDataRepositoryAssociation_disappears_ParentFileSystem
=== PAUSE TestAccFSxDataRepositoryAssociation_disappears_ParentFileSystem
=== RUN   TestAccFSxDataRepositoryAssociation_fileSystemPathUpdated
=== PAUSE TestAccFSxDataRepositoryAssociation_fileSystemPathUpdated
=== RUN   TestAccFSxDataRepositoryAssociation_dataRepositoryPathUpdated
=== PAUSE TestAccFSxDataRepositoryAssociation_dataRepositoryPathUpdated
=== RUN   TestAccFSxDataRepositoryAssociation_importedFileChunkSize
=== PAUSE TestAccFSxDataRepositoryAssociation_importedFileChunkSize
=== RUN   TestAccFSxDataRepositoryAssociation_importedFileChunkSizeUpdated
=== PAUSE TestAccFSxDataRepositoryAssociation_importedFileChunkSizeUpdated
=== RUN   TestAccFSxDataRepositoryAssociation_deleteDataInFilesystem
=== PAUSE TestAccFSxDataRepositoryAssociation_deleteDataInFilesystem
=== RUN   TestAccFSxDataRepositoryAssociation_s3AutoExportPolicy
=== PAUSE TestAccFSxDataRepositoryAssociation_s3AutoExportPolicy
=== RUN   TestAccFSxDataRepositoryAssociation_s3AutoExportPolicyUpdate
=== PAUSE TestAccFSxDataRepositoryAssociation_s3AutoExportPolicyUpdate
=== RUN   TestAccFSxDataRepositoryAssociation_s3AutoImportPolicy
=== PAUSE TestAccFSxDataRepositoryAssociation_s3AutoImportPolicy
=== RUN   TestAccFSxDataRepositoryAssociation_s3AutoImportPolicyUpdate
=== PAUSE TestAccFSxDataRepositoryAssociation_s3AutoImportPolicyUpdate
=== RUN   TestAccFSxDataRepositoryAssociation_s3FullPolicy
=== PAUSE TestAccFSxDataRepositoryAssociation_s3FullPolicy
=== RUN   TestAccFSxLustreFileSystem_basic
=== PAUSE TestAccFSxLustreFileSystem_basic
=== RUN   TestAccFSxLustreFileSystem_disappears
=== PAUSE TestAccFSxLustreFileSystem_disappears
=== RUN   TestAccFSxLustreFileSystem_dataCompression
=== PAUSE TestAccFSxLustreFileSystem_dataCompression
=== RUN   TestAccFSxLustreFileSystem_exportPath
=== PAUSE TestAccFSxLustreFileSystem_exportPath
=== RUN   TestAccFSxLustreFileSystem_importPath
=== PAUSE TestAccFSxLustreFileSystem_importPath
=== RUN   TestAccFSxLustreFileSystem_importedFileChunkSize
=== PAUSE TestAccFSxLustreFileSystem_importedFileChunkSize
=== RUN   TestAccFSxLustreFileSystem_securityGroupIDs
=== PAUSE TestAccFSxLustreFileSystem_securityGroupIDs
=== RUN   TestAccFSxLustreFileSystem_storageCapacity
=== PAUSE TestAccFSxLustreFileSystem_storageCapacity
=== RUN   TestAccFSxLustreFileSystem_storageCapacityUpdate
=== PAUSE TestAccFSxLustreFileSystem_storageCapacityUpdate
=== RUN   TestAccFSxLustreFileSystem_fileSystemTypeVersion
=== PAUSE TestAccFSxLustreFileSystem_fileSystemTypeVersion
=== RUN   TestAccFSxLustreFileSystem_tags
=== PAUSE TestAccFSxLustreFileSystem_tags
=== RUN   TestAccFSxLustreFileSystem_weeklyMaintenanceStartTime
=== PAUSE TestAccFSxLustreFileSystem_weeklyMaintenanceStartTime
=== RUN   TestAccFSxLustreFileSystem_automaticBackupRetentionDays
=== PAUSE TestAccFSxLustreFileSystem_automaticBackupRetentionDays
=== RUN   TestAccFSxLustreFileSystem_dailyAutomaticBackupStartTime
=== PAUSE TestAccFSxLustreFileSystem_dailyAutomaticBackupStartTime
=== RUN   TestAccFSxLustreFileSystem_deploymentTypePersistent1
=== PAUSE TestAccFSxLustreFileSystem_deploymentTypePersistent1
=== RUN   TestAccFSxLustreFileSystem_deploymentTypePersistent2
=== PAUSE TestAccFSxLustreFileSystem_deploymentTypePersistent2
=== RUN   TestAccFSxLustreFileSystem_fromBackup
=== PAUSE TestAccFSxLustreFileSystem_fromBackup
=== RUN   TestAccFSxLustreFileSystem_kmsKeyID
=== PAUSE TestAccFSxLustreFileSystem_kmsKeyID
=== RUN   TestAccFSxLustreFileSystem_deploymentTypeScratch2
=== PAUSE TestAccFSxLustreFileSystem_deploymentTypeScratch2
=== RUN   TestAccFSxLustreFileSystem_storageTypeHddDriveCacheRead
=== PAUSE TestAccFSxLustreFileSystem_storageTypeHddDriveCacheRead
=== RUN   TestAccFSxLustreFileSystem_storageTypeHddDriveCacheNone
=== PAUSE TestAccFSxLustreFileSystem_storageTypeHddDriveCacheNone
=== RUN   TestAccFSxLustreFileSystem_copyTagsToBackups
=== PAUSE TestAccFSxLustreFileSystem_copyTagsToBackups
=== RUN   TestAccFSxLustreFileSystem_autoImportPolicy
=== PAUSE TestAccFSxLustreFileSystem_autoImportPolicy
=== CONT  TestAccFSxDataRepositoryAssociation_basic
=== CONT  TestAccFSxLustreFileSystem_importedFileChunkSize
=== CONT  TestAccFSxLustreFileSystem_deploymentTypePersistent1
=== CONT  TestAccFSxLustreFileSystem_autoImportPolicy
=== CONT  TestAccFSxLustreFileSystem_copyTagsToBackups
=== CONT  TestAccFSxLustreFileSystem_storageTypeHddDriveCacheNone
=== CONT  TestAccFSxLustreFileSystem_storageTypeHddDriveCacheRead
=== CONT  TestAccFSxLustreFileSystem_deploymentTypeScratch2
=== CONT  TestAccFSxLustreFileSystem_kmsKeyID
=== CONT  TestAccFSxLustreFileSystem_fromBackup
--- PASS: TestAccFSxLustreFileSystem_deploymentTypePersistent1 (668.57s)
=== CONT  TestAccFSxLustreFileSystem_deploymentTypePersistent2
--- PASS: TestAccFSxLustreFileSystem_copyTagsToBackups (690.38s)
=== CONT  TestAccFSxDataRepositoryAssociation_s3AutoExportPolicyUpdate
--- PASS: TestAccFSxLustreFileSystem_storageTypeHddDriveCacheRead (710.17s)
=== CONT  TestAccFSxLustreFileSystem_importPath
=== CONT  TestAccFSxDataRepositoryAssociation_s3FullPolicy
--- PASS: TestAccFSxDataRepositoryAssociation_basic (810.24s)
--- PASS: TestAccFSxLustreFileSystem_deploymentTypeScratch2 (814.54s)
=== CONT  TestAccFSxDataRepositoryAssociation_s3AutoImportPolicyUpdate
--- PASS: TestAccFSxLustreFileSystem_storageTypeHddDriveCacheNone (899.64s)
=== CONT  TestAccFSxDataRepositoryAssociation_s3AutoImportPolicy
--- PASS: TestAccFSxLustreFileSystem_deploymentTypePersistent2 (577.11s)
=== CONT  TestAccFSxDataRepositoryAssociation_importedFileChunkSize
--- PASS: TestAccFSxLustreFileSystem_kmsKeyID (1324.42s)
=== CONT  TestAccFSxDataRepositoryAssociation_s3AutoExportPolicy
--- PASS: TestAccFSxLustreFileSystem_fromBackup (1391.03s)
=== CONT  TestAccFSxLustreFileSystem_basic
--- PASS: TestAccFSxLustreFileSystem_autoImportPolicy (1427.43s)
=== CONT  TestAccFSxDataRepositoryAssociation_deleteDataInFilesystem
--- PASS: TestAccFSxDataRepositoryAssociation_s3AutoExportPolicyUpdate (1230.85s)
=== CONT  TestAccFSxLustreFileSystem_exportPath
--- PASS: TestAccFSxDataRepositoryAssociation_importedFileChunkSize (693.10s)
=== CONT  TestAccFSxDataRepositoryAssociation_importedFileChunkSizeUpdated
--- PASS: TestAccFSxLustreFileSystem_importedFileChunkSize (1939.17s)
=== CONT  TestAccFSxLustreFileSystem_dataCompression
--- PASS: TestAccFSxLustreFileSystem_basic (651.82s)
=== CONT  TestAccFSxLustreFileSystem_tags
--- PASS: TestAccFSxDataRepositoryAssociation_s3AutoImportPolicy (1157.67s)
=== CONT  TestAccFSxLustreFileSystem_dailyAutomaticBackupStartTime
--- PASS: TestAccFSxDataRepositoryAssociation_s3FullPolicy (1351.17s)
=== CONT  TestAccFSxLustreFileSystem_automaticBackupRetentionDays
--- PASS: TestAccFSxDataRepositoryAssociation_deleteDataInFilesystem (797.36s)
=== CONT  TestAccFSxLustreFileSystem_weeklyMaintenanceStartTime
--- PASS: TestAccFSxDataRepositoryAssociation_s3AutoImportPolicyUpdate (1552.76s)
=== CONT  TestAccFSxDataRepositoryAssociation_fileSystemPathUpdated
--- PASS: TestAccFSxDataRepositoryAssociation_s3AutoExportPolicy (1189.80s)
=== CONT  TestAccFSxLustreFileSystem_disappears
--- PASS: TestAccFSxLustreFileSystem_importPath (1876.71s)
=== CONT  TestAccFSxDataRepositoryAssociation_disappears_ParentFileSystem
--- PASS: TestAccFSxDataRepositoryAssociation_importedFileChunkSizeUpdated (783.17s)
=== CONT  TestAccFSxLustreFileSystem_storageCapacityUpdate
--- PASS: TestAccFSxLustreFileSystem_dailyAutomaticBackupStartTime (685.21s)
=== CONT  TestAccFSxLustreFileSystem_fileSystemTypeVersion
--- PASS: TestAccFSxLustreFileSystem_tags (728.38s)
=== CONT  TestAccFSxLustreFileSystem_storageCapacity
--- PASS: TestAccFSxLustreFileSystem_dataCompression (837.49s)
=== CONT  TestAccFSxLustreFileSystem_securityGroupIDs
--- PASS: TestAccFSxLustreFileSystem_automaticBackupRetentionDays (700.31s)
=== CONT  TestAccFSxDataRepositoryAssociation_dataRepositoryPathUpdated
--- PASS: TestAccFSxLustreFileSystem_weeklyMaintenanceStartTime (698.33s)
=== CONT  TestAccFSxDataRepositoryAssociation_disappears
--- PASS: TestAccFSxLustreFileSystem_disappears (600.88s)
--- PASS: TestAccFSxDataRepositoryAssociation_fileSystemPathUpdated (823.38s)
--- PASS: TestAccFSxDataRepositoryAssociation_disappears_ParentFileSystem (756.81s)
--- PASS: TestAccFSxDataRepositoryAssociation_disappears (653.33s)
--- PASS: TestAccFSxDataRepositoryAssociation_dataRepositoryPathUpdated (855.43s)
--- PASS: TestAccFSxLustreFileSystem_fileSystemTypeVersion (1245.74s)
--- PASS: TestAccFSxLustreFileSystem_storageCapacity (1254.98s)
--- PASS: TestAccFSxLustreFileSystem_exportPath (2119.46s)
--- PASS: TestAccFSxLustreFileSystem_securityGroupIDs (1307.18s)
--- PASS: TestAccFSxLustreFileSystem_storageCapacityUpdate (2615.37s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/fsx	5342.182s
```

##### GovCloud

```console
% make testacc TESTARGS='-run=TestAccFSxDataRepositoryAssociation_' PKG=fsx ACCTEST_PARALLELISM=10 
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/fsx/... -v -count 1 -parallel 10 -run=TestAccFSxDataRepositoryAssociation_ -timeout 180m
=== RUN   TestAccFSxDataRepositoryAssociation_basic
    data_repository_association_test.go:23: PERSISTENT_2 deployment_type is not supported in GovCloud partition
--- SKIP: TestAccFSxDataRepositoryAssociation_basic (0.00s)
=== RUN   TestAccFSxDataRepositoryAssociation_disappears
    data_repository_association_test.go:62: PERSISTENT_2 deployment_type is not supported in GovCloud partition
--- SKIP: TestAccFSxDataRepositoryAssociation_disappears (0.00s)
=== RUN   TestAccFSxDataRepositoryAssociation_disappears_ParentFileSystem
    data_repository_association_test.go:90: PERSISTENT_2 deployment_type is not supported in GovCloud partition
--- SKIP: TestAccFSxDataRepositoryAssociation_disappears_ParentFileSystem (0.00s)
=== RUN   TestAccFSxDataRepositoryAssociation_fileSystemPathUpdated
    data_repository_association_test.go:119: PERSISTENT_2 deployment_type is not supported in GovCloud partition
--- SKIP: TestAccFSxDataRepositoryAssociation_fileSystemPathUpdated (0.00s)
=== RUN   TestAccFSxDataRepositoryAssociation_dataRepositoryPathUpdated
    data_repository_association_test.go:161: PERSISTENT_2 deployment_type is not supported in GovCloud partition
--- SKIP: TestAccFSxDataRepositoryAssociation_dataRepositoryPathUpdated (0.00s)
=== RUN   TestAccFSxDataRepositoryAssociation_importedFileChunkSize
    data_repository_association_test.go:206: PERSISTENT_2 deployment_type is not supported in GovCloud partition
--- SKIP: TestAccFSxDataRepositoryAssociation_importedFileChunkSize (0.00s)
=== RUN   TestAccFSxDataRepositoryAssociation_importedFileChunkSizeUpdated
    data_repository_association_test.go:240: PERSISTENT_2 deployment_type is not supported in GovCloud partition
--- SKIP: TestAccFSxDataRepositoryAssociation_importedFileChunkSizeUpdated (0.00s)
=== RUN   TestAccFSxDataRepositoryAssociation_deleteDataInFilesystem
    data_repository_association_test.go:281: PERSISTENT_2 deployment_type is not supported in GovCloud partition
--- SKIP: TestAccFSxDataRepositoryAssociation_deleteDataInFilesystem (0.00s)
=== RUN   TestAccFSxDataRepositoryAssociation_s3AutoExportPolicy
    data_repository_association_test.go:314: PERSISTENT_2 deployment_type is not supported in GovCloud partition
--- SKIP: TestAccFSxDataRepositoryAssociation_s3AutoExportPolicy (0.00s)
=== RUN   TestAccFSxDataRepositoryAssociation_s3AutoExportPolicyUpdate
    data_repository_association_test.go:350: PERSISTENT_2 deployment_type is not supported in GovCloud partition
--- SKIP: TestAccFSxDataRepositoryAssociation_s3AutoExportPolicyUpdate (0.00s)
=== RUN   TestAccFSxDataRepositoryAssociation_s3AutoImportPolicy
    data_repository_association_test.go:395: PERSISTENT_2 deployment_type is not supported in GovCloud partition
--- SKIP: TestAccFSxDataRepositoryAssociation_s3AutoImportPolicy (0.00s)
=== RUN   TestAccFSxDataRepositoryAssociation_s3AutoImportPolicyUpdate
    data_repository_association_test.go:431: PERSISTENT_2 deployment_type is not supported in GovCloud partition
--- SKIP: TestAccFSxDataRepositoryAssociation_s3AutoImportPolicyUpdate (0.00s)
=== RUN   TestAccFSxDataRepositoryAssociation_s3FullPolicy
    data_repository_association_test.go:476: PERSISTENT_2 deployment_type is not supported in GovCloud partition
--- SKIP: TestAccFSxDataRepositoryAssociation_s3FullPolicy (0.00s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/fsx	3.674s
```

**Comment by ewbankkit** [comment]
@vicyap Thanks for the contribution :tada: :clap:.

**Pull request merged**